### PR TITLE
refactor: Remove deprecated constructor of TabParagraphContainerComponen

### LINKED ISSUE #7320: Remove deprecated constructor of TabParagraphContainerComponent
https://github.com/SAP/spartacus/blob/develop/projects/storefrontlib/src/cms-components/content/tab-paragraph-container/tab-paragraph-container.component.ts#L51
> `* TODO(issue:#5813) Deprecated since 1.4`

The ticket #5813 is the orginal one, when the deprecation was introduced. So this is a ticket to remove the deprecation as a part of 2.0 cleanup effort.

### PR DESCRIPTION
**Pull request by Zeyber**
Closes: #7320

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/content/tab-paragraph-container/tab-paragraph-container.component.ts [hunk 1]
```diff
@@ -91,14 +76,12 @@ export class TabParagraphContainerComponent
   }
 
   ngOnInit(): void {
-    if (this.winRef && this.winRef.nativeWindow) {
-      const routeState =
-        this.winRef.nativeWindow.history &&
-        this.winRef.nativeWindow.history.state;
+    const routeState =
+      this.winRef.nativeWindow.history &&
+      this.winRef.nativeWindow.history.state;
```
Comment (Pucek9) [inline projects/storefrontlib/src/cms-components/content/tab-paragraph-container/tab-paragraph-container.component.ts]:
```suggestion
      this.winRef.nativeWindow.history?.state;
```

File: projects/storefrontlib/src/cms-components/content/tab-paragraph-container/tab-paragraph-container.component.ts [hunk 2]
```diff
@@ -91,14 +76,12 @@ export class TabParagraphContainerComponent
...
 
-      if (routeState && routeState['activeTab']) {
-        this.activeTabNum = routeState['activeTab'];
-      }
+    if (routeState && routeState['activeTab']) {
```
Comment (Pucek9) [inline projects/storefrontlib/src/cms-components/content/tab-paragraph-container/tab-paragraph-container.component.ts]:
```suggestion
    if (routeState?.['activeTab']) {
```

Comment (Pucek9) [reply projects/storefrontlib/src/cms-components/content/tab-paragraph-container/tab-paragraph-container.component.ts]: or just
```
    const activeTab = this.winRef.nativeWindow.history?.state?.['activeTab'];
    if (activeTab) {
      this.activeTabNum = activeTab;
    }
```

File: projects/storefrontlib/src/cms-components/content/tab-paragraph-container/tab-paragraph-container.component.ts [hunk 3]
```diff
@@ -39,25 +39,10 @@ export class TabParagraphContainerComponent
...
-   *
-   * TODO(issue:#5813) Deprecated since 1.4
-   */
-  constructor(
-    componentData: CmsComponentData<CMSTabParagraphContainer>,
-    cmsService: CmsService
-  );
   constructor(
     public componentData: CmsComponentData<CMSTabParagraphContainer>,
     private cmsService: CmsService,
```
Comment (marlass) [inline projects/storefrontlib/src/cms-components/content/tab-paragraph-container/tab-paragraph-container.component.ts]: Let's make both of these protected as well

**Review by marlass — CHANGES_REQUESTED** [review]

**Review by marlass — APPROVED** [review]

**Pull request merged**